utils/prometheus-node-exporter-lua/files/usr/lib/lua/prometheus-collectors/hostapd_ubus_stations.lua

```diff
@@ -0,0 +1,66 @@
+local ubus = require "ubus"
+local bit32 = require "bit32"
+
+local function get_wifi_interfaces() -- based on hostapd_stations.lua
+  local u = ubus.connect()
+  local status = u:call("network.wireless", "status", {})
+  local interfaces = {}
+
+  for dev, dev_table in pairs(status) do
+    for _, intf in ipairs(dev_table['interfaces']) do
+      table.insert(interfaces, intf['ifname'])
+    end
+  end
+
+  return interfaces
+end
+
+local function scrape()
+  local metric_hostapd_ubus_station_ht = metric("hostapd_ubus_station_ht", "gauge")
+  local metric_hostapd_ubus_station_vht = metric("hostapd_ubus_station_vht", "gauge")
+
+  local metric_hostapd_ubus_station_rrm_caps_link_measurement = metric("hostapd_ubus_station_rrm_caps_link_measurement", "gauge")
+  local metric_hostapd_ubus_station_rrm_caps_neighbor_report = metric("hostapd_ubus_station_rrm_caps_neighbor_report", "gauge")
+  local metric_hostapd_ubus_station_rrm_caps_beacon_report_passive = metric("hostapd_ubus_station_rrm_caps_beacon_report_passive", "gauge")
+  local metric_hostapd_ubus_station_rrm_caps_beacon_report_active = metric("hostapd_ubus_station_rrm_caps_beacon_report_active", "gauge")
+  local metric_hostapd_ubus_station_rrm_caps_beacon_report_table = metric("hostapd_ubus_station_rrm_caps_beacon_report_table", "gauge")
+  
+  local metric_hostapd_ubus_station_rrm_caps_lci_measurement = metric("hostapd_ubus_station_rrm_caps_lci_measurement", "gauge")
+  local metric_hostapd_ubus_station_rrm_caps_ftm_range_report = metric("hostapd_ubus_station_rrm_caps_ftm_range_report", "gauge")
```

Comment:
Would it make sense to define all those outside the scrape function so they are not re-defined on each scrape?

Reply:
nop, see https://github.com/openwrt/packages/blob/master/utils/prometheus-node-exporter-lua/files/usr/bin/prometheus-node-exporter-lua#L45
metric prints the comment line when it's called

Reply:
The other exporter are doing it like this way, too. ;)